src/rgw/rgw_client_io.h

```diff
@@ -332,22 +332,41 @@ class StaticOutputBufferer : public std::streambuf {
  * like rgw::io::Accounter or the AWS Auth v4 stuff implemented by filters
  * while hiding the pipelined architecture from clients.
  *
- * rgw::io::Accounter came in as a part of rgw::io::AccountingFilter. */
-class RGWRestfulIO : public rgw::io::AccountingFilter<rgw::io::RestfulClient*> {
+ * rgw::io::Accounter is injected as a reference to a stage of the pipeline
+ * implementing this interface. */
+class RGWRestfulIO
+  : public rgw::io::DecoratedRestfulClient<rgw::io::RestfulClient*>,
+    public rgw::io::Accounter {
+  /* Reference to a stage of the pipeline that provides the rgw::io::Accounter
+   * interface. */
+  rgw::io::Accounter& accounter;
+
+  /* Just a lifetime holder for dynamically injected stages. */
   std::vector<std::shared_ptr<DecoratedRestfulClient>> filters;
 
 public:
   ~RGWRestfulIO() override = default;
 
-  RGWRestfulIO(CephContext *_cx, rgw::io::RestfulClient* engine)
-    : AccountingFilter<rgw::io::RestfulClient*>(_cx, std::move(engine)) {
+  RGWRestfulIO(CephContext* const ctx,
+               rgw::io::RestfulClient* engine,
+               rgw::io::Accounter* const accounter)
+    : DecoratedRestfulClient<rgw::io::RestfulClient*>(std::move(engine)),
```

Comment:
It looks strange but it necessary as long as the parent class' constructor requires r-value reference. It does that to explicitly inform the next stage never should be accessed directly but with assistance of `DecoratedRestfulClient` instead. I will add a comment.